projects/wave-core-new/src/lib/layers/symbology/symbology.model.ts

```diff
@@ -35,711 +49,733 @@ export const MIN_ALLOWED_POINT_RADIUS = 1;
+    createStyleFunction(): OlStyleFunction {
+        return (feature: OlFeature, _resolution: number): OlStyle => {
+            const styler = this.createStyler(feature);
+
+            const key = styler.cacheKey();
+            if (!(key in styleCache)) {
+                const style = styler.createStyle();
+                styleCache[key] = style;
+                return style;
+            } else {
+                return styleCache[key];
+            }
+        };
     }
 
-    abstract clone(): AbstractSymbology;
-
-    abstract toConfig(): ISymbology;
+    protected abstract createStyler(feature: OlFeature): Styler;
+}
 
-    abstract equals(other: AbstractSymbology): boolean;
+export abstract class Styler {
+    abstract createStyle(): OlStyle;
+    abstract cacheKey(): string;
 
-    abstract toDict(): SymbologyDict;
+    static colorToKey(color: RgbaColor): string {
+        return `${color[0]}${color[1]}${color[2]}${color[3]}`;
+    }
 }
 
-/**
- * Configuration interface with optional fields for VectorSymbology.
- */
-export interface VectorSymbologyConfig extends ISymbology {
-    fillRGBA?: RgbaLike;
-    strokeRGBA?: RgbaLike;
-    strokeWidth?: number;
-    fillColorizer?: IColorizerData;
-    fillColorAttribute?: string;
-    strokeColorizer?: IColorizerData;
-    strokeColorAttribute?: string;
-    strokeDashStyle?: StrokeDashStyle;
-    textAttribute?: string;
-    textColor?: RgbaLike;
-    textStrokeWidth?: number;
-}
+export class PointStyler extends Styler {
+    clustered: boolean;
+    radius: number;
+    fillColor: RgbaColor;
+    stroke: StrokeStyler;
+    text?: TextStyler;
 
-/**
- * Serialzation interface for VectorSymbology.
- */
-interface VectorSymbologyDict extends SymbologyDict {
-    fillRGBA: RgbaTuple;
-    strokeRGBA: RgbaTuple;
-    strokeWidth: number;
-    fillColorizer: IColorizerData;
-    fillColorAttribute: string;
-    strokeColorizer: IColorizerData;
-    strokeColorAttribute: string;
-    strokeDashStyle: StrokeDashStyle;
-    textAttribute: string;
-    textColor: RgbaLike;
-    textStrokeWidth: number;
+    constructor(clustered: boolean, radius: number, fillColor: RgbaColor, stroke: StrokeStyler, text: TextStyler) {
+        super();
+        this.clustered = clustered;
+        this.radius = radius;
+        this.fillColor = fillColor;
+        this.stroke = stroke;
+        this.text = text;
+    }
+
+    createStyle(): OlStyle {
+        const imageStyle = new OlStyleCircle({
+            radius: this.radius,
+            fill: new OlStyleFill({color: this.fillColor}),
+            stroke: this.stroke.createStyle(),
+        });
+
+        return new OlStyle({
+            image: imageStyle,
+            text: this.text ? this.text.createStyle() : undefined,
+        });
+    }
+
+    cacheKey(): string {
+        return `${this.clustered}${this.radius}${Styler.colorToKey(this.fillColor)}${this.stroke.cacheKey()}${
+            this.text ? this.text.cacheKey() : undefined
```

Comment:
```suggestion
            this.text ? this.text.cacheKey() : ''
```
…since you concat strings - or is there a string called undefined?

Reply:
449a8a34ffcfa29f49afba5b97a1188706887ac4